### PR TITLE
Dt 1742

### PR DESCRIPTION
**Pull request by vesameskanen**
The PR is done - it:

- [ ] follows the style and naming rules (passes `npm run lint`)
- [ ] doesn't break anything (passes `npm run test-local` and `npm run test-browserstack`)
- [ ] design is as expected. NOTE! visuals are compared using HSL theme (`CONFIG=hsl npm run dev`).
-- If no design changes: `BS_USERNAME=user BS_ACCESS_KEY=key npm run test-visual` passes
-- If design changes: `BS_USERNAME=user BS_ACCESS_KEY=key npm run test-visual-update` to generate new images
- [ ] any changed files are transformed to ES6
- [ ] all new strings visible to users are

  * [ ] added to translations file
  * [ ] are translated to English, Finnish and Swedish (if not, add translations-needed label to PR)
  
- [ ] all changed components

  * [ ] have examples
  * [ ] are included in the style guide
  * [ ] are included in the visual tests (added to gemini tests)

If this PR fixes a bug, it includes a new test that catches the bug to prevent regressions.

### CODE REVIEW

File: app/client.js [hunk 1]
```diff
@@ -50,13 +51,22 @@ const piwikPlugin = {
...
 
 const ravenPlugin = {
   name: 'RavenPlugin',
   plugContext: plugContext(addRaven),
 };
 
+const addConfig = context =>
+ (context.config = window.config); // eslint-disable-line no-param-reassign
+
+const configPlugin = {
```
Comment (hannesj) [inline app/client.js]: Isn't this done already on the server?

File: app/util/Raven.js [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
 import { COMMIT_ID } from '../buildInfo';
 
-function getRaven() {
-  if (process.env.NODE_ENV === 'production') {
-    /* eslint-disable global-require */
-    const Raven = require('raven-js');
-    Raven.addPlugin(require('raven-js/plugins/console'));
-    /* eslint-enable global-require */
+function getRaven(config) {
+  return function createRaven() {
```
Comment (hannesj) [inline app/util/Raven.js]: Is this level required?

File: app/store/PreferencesStore.js [hunk 3]
```diff
@@ -18,14 +17,18 @@ function loadLanguage() {
 class PreferencesStore extends Store {
   static storeName = 'PreferencesStore';
 
-  language = loadLanguage();
+  constructor(dispatcher) {
+    super(dispatcher);
+    this.config = dispatcher.getContext().config;
```
Comment (hannesj) [inline app/store/PreferencesStore.js]: could we load availableLanguages directly here?

File: app/store/PreferencesStore.js [hunk 4]
```diff
@@ -1,8 +1,7 @@
 import Store from 'fluxible/addons/BaseStore';
 import reactCookie from 'react-cookie';
-import config from '../config';
 
-function loadLanguage() {
+function loadLanguage(config) {
```
Comment (hannesj) [inline app/store/PreferencesStore.js]: this could be moved as an instance method inside the store

File: app/client.js (hunk 1 above)
Comment (vesameskanen) [inline app/client.js]: In the current setup, this makes the config available via fluxible context. Removewthe plugin -> code stops working. Maybe the same functionality can be achieved in some other (simpler?)  way.

File: app/store/PreferencesStore.js (hunk 4 above)
Comment (vesameskanen) [inline app/store/PreferencesStore.js]: Language initialization (2 code lines) moved into constructor.

File: app/util/Raven.js (hunk 2 above)
Comment (vesameskanen) [inline app/util/Raven.js]: Right - config is visible in the client.js where plugin is defined.  Wrapper is indeed redundant, removed.

**Pull request closed without merging**